python/dgl/data/movielens.py

```diff
@@ -0,0 +1,640 @@
+    - Ratings: 100,000 (1, 2, 3, 4, 5)
+
+    MovieLens-1M (ml-1m)
+
+    - Users: 6,040
+    - Movies: 3,706
+    - Ratings: 1,000,209 (1, 2, 3, 4, 5)
+
+    MovieLens-10M (ml-10m)
+
+    - Users: 69,878 
+    - Movies: 10,677
+    - Ratings: 10,000,054 (0.5, 1, 1.5, ..., 4.5, 5.0)
+
+    Parameters
+    ----------
+    name: str
+        Dataset name. (:obj:`"ml-100k"`, :obj:`"ml-1m"`, :obj:`"ml-10m"`). 
+    valid_ratio: int
+        Ratio of validation samples out of the whole dataset. Should be in (0.0, 1.0).
+    test_ratio: int
+        Ratio of testing samples out of the whole dataset. Should be in (0.0, 1.0) for ml-1m and ml-10m. This parameter is invalid
+        when :obj:`name` is :obj:`"ml-100k"`, since it testing samples are pre-specified.
+        Default: None
+    raw_dir : str
+        Raw file directory to download/contain the input data directory.
+        Default: ~/.dgl/
+    force_reload : bool
+        Whether to reload the dataset. Default: False
+    verbose : bool
+        Whether to print out progress information. Default: True.
+    reverse_edge : bool
+        Whether to add reverse edges in graph. Default: True.
+    transform : callable, optional
+        A transform that takes in a :class:`~dgl.DGLGraph` object and returns
+        a transformed version. The :class:`~dgl.DGLGraph` object will be
+        transformed before every access.
+
+    Attributes
+    ----------
+    train_graph: dgl.DGLGraph
+        Training graph
+    valid_graph: dgl.DGLGraph
+        Validation graph
+    test_graph: dgl.DGLGraph
+        Testing graph
+    user_feat: torch.Tensor
+        User features
+    movie_feat: torch.Tensor
+        Movie features
+    info: dict
+        Store training, validation and testing rating pairs.
+
+    Notes
+    -----
+    - The number of edges is doubled to form an undirected(bidirected) graph structure.
+    - Each time when the dataset is loaded with a different setting of training, validation and testing ratio, the param
+    :obj:`"force_reload"` should be set to :obj:`"True"`. Otherwise the previous split of dataset will be loaded and not be
+    updated in the storage.
+
+    Examples
+    --------
+    >>> from dgl.data import MovieLensDataset
+    >>> 
+    >>> dataset = MovieLensDataset(name='ml-100k', valid_ratio=0.2)
+    >>> train_g, valid_g, test_g = dataset[0]
+    >>> train_g
+    Graph(num_nodes=2625, num_edges=128000,
+        ndata_schemes={}
+        edata_schemes={'_ID': Scheme(shape=(), dtype=torch.int64), 'etype': Scheme(shape=(), dtype=torch.float32)})
+    >>>
+    >>> # get ratings of edges in the training graph
+    >>> ratings = train_g.edata['etype']
+    >>> ratings
+    tensor([3., 3., 2.,  ..., 4., 4., 2.])
+    >>>
+    >>> # get training, validation and testing rating pairs
+    >>> train_rating, valid_rating, test_rating = \
+    ...     dataset.info['train_rating_pairs'], dataset.info['valid_rating_pairs'], dataset.info['test_rating_pairs']
```

Comment:
What's the shape of train_rating_pairs? What are the semantic meanings of the dimensions?